### PR TITLE
DAF-4308: Fix black screen shows up for a moment when back from PIP

### PR DESCRIPTION
**Pull request by ghost**
Because the current logic seems to still trigger the `_betterPlayerController.disablePictureInPicture` even when the user presses the `back to app` button in PIP mode (which causes bug DAF-4308), I will emit the `isBackToAppButtonPressed` value to the Flutter side and handle it there. I will also delete the unused logic.

https://dw-ml-nfc.atlassian.net/browse/DAF-4308

### CODE REVIEW

File: ios/Classes/BetterPlayer.m [hunk 1]
```diff
@@ -783,10 +774,8 @@ - (void)pictureInPictureController:(AVPictureInPictureController *)pictureInPict
 
 - (void)pictureInPictureController:(AVPictureInPictureController *)pictureInPictureController restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:(void (^)(BOOL))completionHandler {
     [self setRestoreUserInterfaceForPIPStopCompletionHandler: true];
-    if (_isDisablePIPManually) {
-        _isDisablePIPManually = false;
-    } else {
-        _isBackToAppButtonPressed = true;
+    if (_eventSink != nil) {
```
Comment (mvn-thanhluu-hn) [inline ios/Classes/BetterPlayer.m:777]: memo:
- With this implementation, we will handle logic open `from PIP return button` or `manually` completely on flutter side.

**Review by mvn-thanhluu-hn — APPROVED** [review]
LGTM.

File: ios/Classes/BetterPlayer.h [hunk 2]
```diff
@@ -38,8 +38,6 @@ NS_ASSUME_NONNULL_BEGIN
 @property(nonatomic) float playerRate;
 @property(nonatomic) int overriddenDuration;
 @property(nonatomic) AVPlayerTimeControlStatus lastAvPlayerTimeControlStatus;
-@property(nonatomic) bool isBackToAppButtonPressed;
```
Comment (noboru-i) [inline ios/Classes/BetterPlayer.h:41]: memo
This variable is added on https://github.com/dwango-nfc/betterplayer/pull/42/files#diff-a2f4ffe7e3d54263623c9291e48051d1a11bd712ca7ca05d00cee0338e619598R41
It is related to DAF-4278.

File: ios/Classes/BetterPlayer.h [hunk 3]
```diff
@@ -38,8 +38,6 @@ NS_ASSUME_NONNULL_BEGIN
...
-@property(nonatomic) bool isDisablePIPManually;
```
Comment (noboru-i) [inline ios/Classes/BetterPlayer.h:42]: memo
This variable is added on https://github.com/dwango-nfc/betterplayer/pull/43/files#diff-a2f4ffe7e3d54263623c9291e48051d1a11bd712ca7ca05d00cee0338e619598R42
It is related to DAF-4278.

File: lib/src/video_player/method_channel_video_player.dart [hunk 4]
```diff
@@ -493,6 +482,11 @@ class MethodChannelVideoPlayer extends VideoPlayerPlatform {
             eventType: VideoEventType.failedToPlayToEndTime,
             key: key,
           );
+        case 'isBackToAppButtonPressed':
```
Comment (noboru-i) [inline lib/src/video_player/method_channel_video_player.dart]: `is` prefix is usually used as "function that returns bool" or "bool variable name".
How about renaming this and the same names to `tapBackToAppButton`? (or `pressedBackToAppButton`, etc..)

**Review by noboru-i — COMMENTED** [review]
@mvn-datnguyen-hn 
Please check my comment 🙏 

@HiroshiOshiro 
Please review this :bow:

**Review by noboru-i — APPROVED** [review]
LGTM.

Please wait @HiroshiOshiro san's review.

**Review by HiroshiOshiro — APPROVED** [review]
LGTM.

**Pull request merged**